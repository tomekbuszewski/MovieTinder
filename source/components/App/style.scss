@import url('https://fonts.googleapis.com/css?family=Muli:200,800|Roboto:300');
@import '_normalize';

@import '_globals';

* { box-sizing: border-box; }

html {
  font-family: "Roboto", "Helvetica", sans-serif;
  font-size: 8px;
}

body {
  overflow: hidden;
  height: 100vh;
  background: $Mirage;
}
