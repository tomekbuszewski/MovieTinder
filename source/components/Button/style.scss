@import '../App/globals';

$Button: '.Button';

#{$Button} {
  width: 100%;
  height: 5rem;
  border: 0;
  background: none;
  border-radius: 2.5rem;
  color: $PotPourri;
  cursor: pointer;
  font-family: $fontSecondary;
  font-size: map_get($fonts, text);
  font-weight: 800;
  line-height: 4.5rem;
  outline: 0;
  text-align: center;

  &--Accept {
    background: linear-gradient(45deg, $Matisse 0%, $Bossanova 100%);
  }
  &--Decline {
    width: 14rem;
    border: 0.25rem solid $Bossanova;
    margin-right: 2rem;
  }

  @media (min-width: map_get($mq, md)) {

    &--Decline {
      width: 100%;
      order: 1;
      margin: 2rem 0 0;
    }
  }
}
