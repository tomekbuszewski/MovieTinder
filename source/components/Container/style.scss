.container {
  width: 100%;
  margin: auto;

  @media (min-width: 768px) {
    max-width: 740px;
  }

  @media (min-width: 1024px) {
    max-width: 900px;
  }

  @media (min-width: 1240px) {
    max-width: 1200px;
  }
}
