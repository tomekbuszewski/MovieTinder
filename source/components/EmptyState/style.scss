@import '../App/globals';

.EmptyState {
  position: absolute;
  top: 50%;
  left: 50%;
  color: $PotPourri;
  text-align: center;
  transform: translate(-50%, -50%);

  &__Text {
    margin-top: 3rem;
  }
}
