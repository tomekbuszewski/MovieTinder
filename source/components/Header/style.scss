.header {
  padding: 20px 10px;

  @media (min-width: 768px) {
    padding: 40px 20px;
  }

  @media (min-width: 1280px) {
    padding: 80px 20px;
  }
}

.logo {
  font-size: 36px;
  color: #737373;
  text-transform: uppercase;

  span {
    color: #fdc22a;
  }
}
