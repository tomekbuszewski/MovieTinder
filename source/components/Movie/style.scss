/**
Reused my old code from https://codepen.io/tomekbuszewski/pen/RaQvem
 */

@import '../App/globals';

$initialTransform: 10px;
$loop: 20;

$Movie: 'Movie';

.#{$Movie} {
  text-align: center;
  width: 100%;
  max-width: 32rem;
  height: 75vh;
  margin-top: 10vh;
  background: #fafafa;
  border-radius: 2rem;
  box-shadow: 0 2rem 3rem 0 rgba(0, 0, 0, 0.1);
  position: absolute;
  top: 0;
  left: 50%;
  transition: 0.5s transform, 0.5s opacity;
  display: flex;
  overflow: hidden;
  pointer-events: none;
  flex-flow: row wrap;
  align-items: center;
  padding: 2rem;

  &:last-of-type { pointer-events: all; }

  @for $i from $loop through 1 {
    &:nth-last-of-type(#{$i}) {
      $transform: ($i) * $initialTransform;
      $scale: 1 - ($i / 100);

      transform: translateY($transform) translateX(-50%) scale($scale);
    }
  }

  &--Accepted:nth-of-type(n) {
    transform: translate(-50%, -10rem);
    opacity: 0;
  }

  &--Rejected:nth-of-type(n) {
    transform: translate(25rem, 4rem) rotate(20deg);
    opacity: 0;
  }

  &__Buttons {
    position: absolute;
    bottom: 0;
    left: 0;
    width: 100%;
    display: flex;
    padding: 2rem;
  }

  &__Button {
    font-size: 1.2rem;
    font-weight: bold;
    padding: 1rem 2rem;
    border: 0;
    border-radius: 3rem;

    &--Accept {
      background: map_get($colors, green);
      width: 75%;
      margin-left: 1rem;
    }

    &--Decline {
      background: map_get($colors, red);
      color: map_get($colors, white);
    }
  }
}
