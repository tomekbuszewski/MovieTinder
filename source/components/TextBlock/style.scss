@import '../App/globals';

.TextBlock {
  margin: 0;
  color: $PotPourri;
  font-size: map_get($fonts, text);
  font-weight: 300;
  line-height: 1.5;

  @media (min-width: map_get($mq, sm)) and (max-width: map_get($mq, md) - 1) {
    font-size: map_get($fonts, small);
  }
}
