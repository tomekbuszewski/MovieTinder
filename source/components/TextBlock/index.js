import React from 'react';
import PropTypes from 'prop-types';

import style from './style.scss';

// @flow
const TextBlock = ({input, additionalClassNames, truncate}: {input: string, additionalClassNames: string, truncate: number}) => {
  const text = truncate === 0 ? input : `${input.substr(0, truncate)}...`;

  return <p className={`${style['TextBlock']} ${additionalClassNames}`}>{text}</p>;
};

TextBlock.propTypes = {
  input: PropTypes.string,
  additionalClassNames: PropTypes.string,
  truncate: PropTypes.number
};

TextBlock.defaultProps = {
  additionalClassNames: '',
  truncate: 0
};

export default TextBlock;
