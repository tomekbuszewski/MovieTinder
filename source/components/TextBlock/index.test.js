import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

import TextBlock from './index';
import style from './style.scss';

describe ('Text Block', () => {
  it ('should match snapshot', () => {
    const component = renderer.create(<TextBlock input="Some text" />).toJSON();

    expect(component).toMatchSnapshot();
  });

  it ('should display proper text', () => {
    const component = shallow(<TextBlock input="Some text" />);

    expect(component.text()).toEqual('Some text');
  });

  it ('should contain default class name', () => {
    const component = mount(<TextBlock input="Some text" />);

    expect(component.find('p').hasClass(style['TextBlock'])).toBeTruthy();
  });

  it ('can have more classes added', () => {
    const component = mount(<TextBlock input="Some text" additionalClassNames="test class" />);

    expect(component.find('p').hasClass('test class')).toBeTruthy();
  });
});
