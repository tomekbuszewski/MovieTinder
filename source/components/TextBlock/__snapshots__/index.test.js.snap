// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Block should match snapshot 1`] = `
<p
  className="TextBlock "
>
  Some text
</p>
`;
